cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/PersonDrawWidget.kt

```diff
@@ -0,0 +1,121 @@
+package com.cognifide.cogboard.widget.type
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.config.service.BoardsConfigService
+import com.cognifide.cogboard.storage.ContentRepository
+import com.cognifide.cogboard.widget.BaseWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import kotlin.random.Random
+
+class PersonDrawWidget(vertx: Vertx, config: JsonObject, boardService: BoardsConfigService = BoardsConfigService()) :
+    BaseWidget(vertx, config, boardService) {
+
+    private val valuesArch: MutableList<String> = mutableListOf()
+    private var updateDate: LocalDateTime = LocalDateTime.now()
+    private var currentIndex: Int = -1
+    private var usedIndexes: MutableList<Int> = mutableListOf()
+
+    init {
+        super.config.put(CogboardConstants.PROP_SCHEDULE_PERIOD, SYNC_INTERVAL)
+        if (currentIndex == -1) {
+            currentIndex = ContentRepository().get(id).getInteger(PROP_CONTENT_INDEX, -1)
+            val updateDateMillis = ContentRepository().get(id).getLong(PROP_CONTENT_UPDATE_DATE, -1)
+            if (updateDateMillis > 0) {
+                val ofEpochMilli = Instant.ofEpochMilli(updateDateMillis)
+                updateDate = ofEpochMilli.atZone(ZoneId.systemDefault()).toLocalDateTime()
+            }
+        }
+
+        createDynamicChangeSubscriber()
+                .handler { cycle(forceCycle = true) }
+    }
+
+    override fun updateState() {
+        cycle()
+    }
+
+    private fun cycle(forceCycle: Boolean = false) {
+        val isDaily = config.getBoolean(PROP_IS_DAILY, false)
+        val values = toStringList(config.getJsonArray(PROP_VALUES, JsonArray()))
+        val randomize = config.getBoolean(PROP_RANDOMIZE, false)
+        val interval = config.getLong(PROP_INTERVAL, SELECT_INTERVAL)
+
+        if (!isDaily && updateDate.isAfter(LocalDateTime.now().plusMinutes(interval))) {
+            updateDate = LocalDateTime.now()
+        }
+
+        if (forceCycle || shouldCycle(currentIndex, updateDate) || values != valuesArch) {
+            syncValues(values)
+            currentIndex = newIndex(randomize, currentIndex, values, usedIndexes)
+            updateDate = calculateNextUpdateDate(isDaily, interval)
+            sendUpdate(currentIndex, updateDate)
+        }
+    }
+
+    private fun syncValues(values: MutableList<String>) {
+        valuesArch.clear()
+        valuesArch.addAll(values)
+    }
+
+    private fun toStringList(values: JsonArray): MutableList<String> {
+        val list = mutableListOf<String>()
```

Comment:
in kotlin it can be even shorter I believe

```
        values.stream()
                .filter { it is String }
                .map { it as String }
```

Reply:
or even

```
val list = values.filterIsInstance<String>()
```